samples-final/8.AspNetCore-LUIS-Bot/Startup.cs

```diff
@@ -65,14 +54,71 @@ public void ConfigureServices(IServiceCollection services)
-                var model = new LuisModel(modelId, subscriptionKey, url);
-                options.Middleware.Add(new LuisRecognizerMiddleware(model));
+                // Forces storage to auto-load on a new message, and auto-save when complete.  
+                // Put at the beginning of the pipeline.
+                var stateSet = new BotStateSet(options.State.ToArray());
+                options.Middleware.Add(stateSet);                                                     
+            });
+
+            // Now that the bot is registered, create and register any state accesssors. 
+            // These accessors are passed into the Bot on every turn. 
+            services.AddSingleton<MyBotAccessors>(sp =>
+            {
+                var options = sp.GetRequiredService<IOptions<BotFrameworkOptions>>().Value;
+                if (options == null)
+                {
+                    throw new InvalidOperationException("BotFrameworkOptions must be configured prior to setting up the State Accessors");
+                }
+
+                var userState = options.State.OfType<UserState>().FirstOrDefault();
+                if (userState == null)
+                {
+                    throw new InvalidOperationException("UserState must be defined and added before adding user-scoped state accessors.");
+                }
+
+                var accessors = new MyBotAccessors
+                {
+                    Reminders = userState.CreateProperty<List<Reminder>>(MyBotAccessors.RemindersName, () => new List<Reminder>()),
+                    UserDialogState = userState.CreateProperty<Dictionary<string, object>>(MyBotAccessors.DialogStateName, () => new Dictionary<string, object>())
+                };
+
+                return accessors;
+            });
+
+            // Create a LUIS Recognizer that is initialized and suitable for passing
+            // into the Bot on each turn. 
+            services.AddSingleton<LuisRecognizer>(sp =>
+            {
+                var applicationId = Configuration.GetSection("Luis-ApplicationId")?.Value;
+                var endpointKey = Configuration.GetSection("Luis-EndpointKey")?.Value;
+                var azureRegion = Configuration.GetSection("Luis-AzureRegion")?.Value;
+
+                if (string.IsNullOrWhiteSpace(applicationId))
+                {
+                    throw new InvalidOperationException("The Luis ApplicationId ('Luis-ApplicationId') is required to run this sample.");
+                }
+
+                if (string.IsNullOrWhiteSpace(endpointKey))
+                {
+                    throw new InvalidOperationException("The Luis endpoint key ('Luis-EndpointKey') is required to run this sample.");
+                }
+
+                if (string.IsNullOrWhiteSpace(azureRegion))
+                {
+                    throw new InvalidOperationException("The Luis Azure Region ('Luis-AzureRegion') is required to run this sample.");
```

Comment:
This is cool. For the benefit of a newbie developer that just gets this sample and hits F5, can we have this appended to the exception text - "Please update your appsettings.json."

Reply:
Fixed, good idea